demos/common/mqtt/CMakeLists.txt

```diff
@@ -53,3 +53,14 @@ afr_module_dependencies(
     INTERFACE
         AFR::mqtt
 )
+
+# MQTT-over-BLE demo.
+# Only build on boards that support BLE.
+if(${BLE_SUPPORTED})
```

Comment:
You don't need to guard this, just make it depends on `AFR::ble` and it will be automatically disabled when `AFR::ble` is not available.